replica/compaction_group.hh

```diff
@@ -62,7 +68,11 @@ private:
 
     future<> delete_sstables_atomically(std::vector<sstables::shared_sstable> sstables_to_remove);
 public:
-    compaction_group(table& t, dht::token_range token_range);
+    compaction_group(table& t, idx_type idx, dht::token_range token_range);
+
+    idx_type idx() const noexcept {
+        return _idx;
```

Comment:
perhaps idx_type index()? I know we're well familiar with what idx stands for, but full name leaves no doubt. e.g. google tells me it stands for "Internet Data Exchange" :-)

Reply:
This is some bikeshedding from me though, feel free to ignore it.

Reply:
changing that in next version to present a fully fledged `struct group_id` that contains both idx and total.
I think it makes more sense, once exposed also for table state